discovery_engine/lib/src/domain/engine/mock_engine.dart

```diff
@@ -43,6 +43,11 @@ class MockEngine implements Engine {
         'name': 'domain',
         'thumbnail': 'http://thumbnail.domain.com',
       },
+      'rank': 10,
+      'score': 0.1,
+      'country': 'en',
```

Comment:
`country` would always be in all-caps (ISO 3166-1 alpha-2).

Reply:
done. Do you why it should be all capital? Internally we can probably relax this.

Reply:
My guess is they're specified all-capital because the country codes have different semantics from the language codes. For example, the ISO 639-1 language code `cy` means Welsh, whereas the ISO 3166-1 alpha-2 country code `CY` means Cyprus. Later on we may want to use an `enum` internally, to avoid any confusion on the matter ^^

Reply:
I'll prefer a new type with a check in the constructor instead of listing all the codes. Hopefully we will find a library already support this format.

Reply:
I don't think I understand. What's the difference between having a library that uses an `enum` with a `from_string` vs. your proposed solution? The crates that I've skimmed earlier all internally represented their states as enums. Is the aversion to enums related to the FFI?

Reply:
That the library has already all the definitions of the languages and code to check if is a valid one and we don't need to write it. If the lib use an enum is even better.